espnet/nets/batch_beam_search_online_sim.py

```diff
@@ -0,0 +1,254 @@
+"""Parallel beam search module for online simulation."""
+
+import logging
+from pathlib import Path
+from typing import List
+
+import yaml
+
+import torch
+
+from espnet.nets.batch_beam_search import BatchBeamSearch
+from espnet.nets.beam_search import Hypothesis
+from espnet.nets.e2e_asr_common import end_detect
+
+
+class BatchBeamSearchOnlineSim(BatchBeamSearch):
```

Comment:
What does "Sim" come from?
Simulation? (imitating the online mode?)
It's better to add some description about it.